src/sentry/utils/registry.py

```diff
@@ -15,9 +15,10 @@ class NoRegistrationExistsError(ValueError):
 
 
 class Registry(Generic[T]):
-    def __init__(self):
+    def __init__(self, enable_reverse_lookup: bool = True):
```

Comment:
needed to add this to disable reverse lookup for the new registry. there may be multiple detectors with the same config schema (e.g. empty config), and enabling reverse lookup means keys AND values in the registry must be unique.